nixos/modules/services/misc/transfer-sh.nix

```diff
@@ -0,0 +1,434 @@
+        default = "${cfg.stateDir}/transfer-sh.log";
+        type = types.nullOr types.str;
+      };
+
+      CORS_DOMAINS = mkOption {
+        description = "comma separated list of domains for CORS, setting it enable CORS";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      CLAMAV_HOST = mkOption {
+        description = "host for clamav feature";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      PERFORM_CLAMAV_PRESCAN = mkOption {
+        description = "prescan every upload through clamav feature (clamav-host must be a local clamd unix socket)";
+        default = false;
+        type = types.nullOr types.bool;
+      };
+
+      RATE_LIMIT = mkOption {
+        description = "request per minute";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      MAX_UPLOAD_SIZE = mkOption {
+        description = "max upload size in kilobytes";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      PURGE_DAYS = mkOption {
+        description = "number of days after the uploads are purged automatically";
+        default = "7";
+        type = types.nullOr types.str;
+      };
+
+      PURGE_INTERVAL = mkOption {
+        description = "interval in hours to run the automatic purge for (not applicable to S3 and Storj)";
+        default = 1;
+        type = types.nullOr types.int;
+      };
+
+      RANDOM_TOKEN_LENGTH = mkOption {
+        description = "length of the random token for the upload path (double the size for delete path)";
+        default = "6";
+        type = types.nullOr types.str;
+      };
+
+      stateDir = mkOption {
+        type = types.path;
+        description = "Variable state directory";
+        default = "/var/lib/transfer.sh";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable
+    {
+      users.users = mkIf (cfg.user == "transfersh") {
+        transfersh = {
+          description = "transfer-sh service user";
+          home = cfg.stateDir;
+          group = cfg.group;
+          isSystemUser = true;
+        };
+      };
+
+      users.groups = mkIf (cfg.group == "transfersh") { transfersh = { }; };
+
+      systemd.tmpfiles.rules = [
+        "d ${cfg.stateDir} 0750 ${cfg.user} ${cfg.group} - -"
+        "d ${cfg.BASEDIR} 0750 ${cfg.user} ${cfg.group} - -"
+      ];
+
+      systemd.services.transfer-sh = {
+        wantedBy = [ "multi-user.target" ];
+        after = [ "network.target" ];
+        serviceConfig = {
+          User = cfg.user;
+          Group = cfg.group;
+          ExecStart = "${cfg.package}/bin/transfer.sh --provider=${cfg.provider}";
+          EnvironmentFile = lib.mkIf (cfg.environmentFile != null) cfg.environmentFile;
+        };
+
+        environment =
+          let
+            boolString = s: if s then "true" else "false";
+          in
+          {
+            LISTENER = "${cfg.address}:${builtins.toString cfg.LISTENER}";
```

Comment:
You can call `toString` without `builtins.`

Reply:
deleted